gaiku-common/src/texture.rs

```diff
@@ -129,6 +130,7 @@ where
     // Convert uv to tex xy for the origin of this blit
     let x_o = (uv[0] * self.texture.width() as f32).floor() as u32; // Convert uv to tex xy
     let y_o = (uv[1] * self.texture.height() as f32).floor() as u32; // Convert uv to tex xy
+
     // TODO: We don't need to invert the y?
     // let y_o = ((1.0 - COL_SIZE - uv.1) * self.texture.height() as f32).floor() as u32; // Convert uv to tex xy and invert the Y axis, so it starts from top
```

Comment:
What do you think about it, I really didn't inspect it, while the amethyst and bevy examples shown the texture correctly I didn't check in blender, just want to be sure that it works correctly, but sadly today I'm out of steam to check that.

Reply:
This all dependes on the input and how we define the coordinates. If we invert the `v` should we invert the all the pixels from the user for that tile. If we do that then the pixels passed from the users will be upside down in the image which might confuse them.

Or are you just talking about shifting the tile start coodinate? Currently we use the one from the `self.get_uv` method (because that method has the calculations for the padding and I don't want to duplicate that code in here). We take the first of the four coordinates.

Which comes from this approximate method:
```rust
let (x,y) = (index % COLS as u8, index / COLS as u8);
let (u, v) = (x as f32 / COLS as f32, (ROWS as u8 - 1 - y) as f32 / ROWS as f32,);

let uv_1 = [u, v];
let uv_2 = [u + COL_SIZE, v];
let uv_3 = [u  + COL_SIZE, v + ROW_SIZE];
let uv_4 = [u, v + ROW_SIZE];
```

With `index = 0`
```rust
let index = 0;
let (x,y) = (0 % 16, 0 / 16) = (0, 0);
let (u, v) = (0/ 16, 16 - 1 - 0) / 16) = (0, 0.938);

let uv_1 = [0, 0.938];
let uv_2 = [0 + 1/16, 0.938] = [0.063, 0.938];
let uv_3 = [0  + 1/16, 0.938 + 1/16]  = [0.063, 1.];
let uv_4 = [0, 0.938 + 1/16]  = [0.063, 1.];
```

This corresponds to the uvs being in these places:

```
------------
|  1     2 |
|  3     4 |
------------
```

Since we are blitting left to right top to bottom we should not invert this coordinate. Unless we want to blit from bottom to top and invert the users image.

Reply:
Blender has inverted v axis. When exporting to blender we can invert the v on save. (Or in blender itself you can flip the coordinate by applying a bit of math to the nodes)

Reply:
Btw why do we start the tiles from the bottom left and work upwards?

Reply:
> Blender has inverted v axis. When exporting to blender we can invert the v on save. (Or in blender itself you can flip the coordinate by applying a bit of math to the nodes)

Ok, that's why the textures looked off in blender.

> Btw why do we start the tiles from the bottom left and work upwards?

Was a side effect on start from `0,0`, there's no specific reason, but we can change that.